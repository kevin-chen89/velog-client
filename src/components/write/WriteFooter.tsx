import * as React from 'react';
import styled, { css } from 'styled-components';
import Button from '../common/Button';
import { WriteMode } from '../../modules/write';

const WriteFooterBlock = styled.div`
  padding-left: 1rem;
  padding-right: 1rem;
  height: 4rem;
  width: 100%;
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
  background: rgba(255, 255, 255, 0.85);
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

const StyledButton = styled(Button)`
  height: 2.5rem;
  font-size: 1.125rem;
  & + & {
    margin-left: 0.75rem;
  }
`;

interface WriteFooterProps {}

const WriteFooter: React.FC<WriteFooterProps> = props => {
  return (
    <WriteFooterBlock>
      <StyledButton inline color="gray">
        임시저장
      </StyledButton>
      <StyledButton inline color="teal">
        출간하기
      </StyledButton>
    </WriteFooterBlock>
  );
};

export default WriteFooter;
