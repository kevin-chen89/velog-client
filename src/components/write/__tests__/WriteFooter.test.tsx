import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';
import WriteFooter, { WriteFooterProps } from '../WriteFooter';

describe('WriteFooter', () => {
  const setup = (props: Partial<WriteFooterProps> = {}) => {
    const initialProps: WriteFooterProps = {
      onPublish: () => {},
      onTempSave: () => {},
    };
    const utils = render(<WriteFooter {...initialProps} {...props} />);
    const buttons = {
      tempSave: utils.getByText('임시저장'),
      publish: utils.getByText(/(출간|수정)하기/),
    };
    return {
      ...utils,
      buttons,
    };
  };
  it('renders properly', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('buttons are working properly', () => {
    const onPublish = jest.fn();
    const onTempSave = jest.fn();
    const utils = setup({ onPublish, onTempSave });
    const { tempSave, publish } = utils.buttons;
    fireEvent.click(tempSave);
    expect(onTempSave).toBeCalled();
    fireEvent.click(publish);
    expect(onPublish).toBeCalled();
  });
  it('changes button text when edit is true', () => {
    const { buttons } = setup({
      edit: true,
    });
    expect(buttons.publish.textContent).toBe('수정하기');
  });
});
