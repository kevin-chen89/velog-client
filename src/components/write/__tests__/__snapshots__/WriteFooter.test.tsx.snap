// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WriteFooter matches snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH cvCbtW"
  >
    <button
      class="sc-htpNat bGXGEN sc-bdVaJa eLsWnY"
      color="lightGray"
    >
      임시저장
    </button>
    <button
      class="sc-htpNat bGXGEN sc-bdVaJa jejcDj"
      color="teal"
    >
      출간하기
    </button>
  </div>
</div>
`;
